### ISSUE TITLE
Mark pure reformatting PRs in .git-blame-ignore-revs

### ISSUE THREAD

**Issue by jakkdl** [python-trio/trio#2837]
This is mostly relevant to @CoolCat467 as you're in the progress of cleaning up the repository, but I realize we in general haven't been great at keeping that file updated.

See discussion in https://github.com/python-trio/trio/pull/2686 and #2962

It would be great if one could have a check that warned about this ... but I have no clue how that would work in any way, and I'm also not sure if it's possible to learn the commit ID's before merging such that you can actually add a PR to the file before it gets merged.

**Comment by A5rocks** [comment]
> I'm also not sure if it's possible to learn the commit ID's before merging such that you can actually add a PR to the file before it gets merged.

I don't think so, because it'll use the current timestamp. Is there anything you think should get ignored? I looked at the blame for the abc file and the only ones I see are at least 4 years old.

**Comment by jakkdl** [comment]
> > I'm also not sure if it's possible to learn the commit ID's before merging such that you can actually add a PR to the file before it gets merged.
> 
> I don't think so, because it'll use the current timestamp.

I feel like it should be possible with some post-merge github action or something, but probs not worth setting up anything overly complicated.

> Is there anything you think should get ignored? I looked at the blame for the abc file and the only ones I see are at least 4 years old.

Did a quick look and there's a bunch of recent PR's that are somewhat borderline, like #2930 and #2932. #2823 has a mix of blocks of reformatting and other stuff.
Not sure about #3024. #3044 definitely merits it.

If we *really* wanted to we could split partial-reformatting PR's into two commits that we merge onto master, ignoring one of them, instead of squashing.

**Comment by jakkdl** [comment]
https://github.com/python-trio/trio/commit/acdb7b2c424a2f91f8f98e63010e423b2dc49358 should be added imo. It's type-checking and not pure formatting, but I don't think it'll ever be useful to hit in a `git blame` (as I just did)
